### PR TITLE
Fix tsconfig in `docs` example

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

- Added `"types": ["vite/client"]` because previosly I was getting rid squigglies everywhere an environment variable was used.
- Moved `moduleResolution` under `compilerOptions`. Previously it had no effect.

## Testing

Tested manually / in IDE.

## Docs

No change in main codebase but it will affect `create-astro` (I think). So it's "visible" in that sense. Not sure if I should still run `pnpm exec changeset`.

### CODE REVIEW

File: examples/docs/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "compilerOptions": {
     "target": "es2020",
     "module": "esnext",
-    "jsx": "preserve"
+    "jsx": "preserve",
+    "moduleResolution": "node",
+    "types": ["astro/env"]
```
Comment (Princesseuh) [inline examples/docs/tsconfig.json:7]: For `import.meta.env` support inside non-Astro files, it's better to use `"types": ["vite/client"]` than `astro/env`. 

Using `astro/env`, makes the Astro global available in non-Astro files, which doesn't work (at the moment)

**Comment by Princesseuh** [comment]
Just had a small comment on the usage of `astro/env` instead of `vite/config` but everything else is good!

File: examples/docs/tsconfig.json (hunk 1 above)
Comment (mayank99) [reply examples/docs/tsconfig.json]: Yup, that's what I had before but then I found the docs which suggested using `astro/env`. https://docs.astro.build/en/guides/typescript/#setup

Changed.

**Comment by JuanM04** [comment]
I think this is being fixed by #3107

**Comment by mayank99** [comment]
> I think this is being fixed by #3107

Oh, my bad. Should have searched before creating this PR. Good to see it's being made consistent across all examples 👍 

Also might be important to discuss `vite/client` vs `astro/env` in that PR then.

**Comment by JuanM04** [comment]
> Also might be important to discuss vite/client vs astro/env in that PR then.

`astro/env` includes `vite/client` plus some other Astro envs: https://github.com/withastro/astro/blob/main/packages/astro/env.d.ts. I think that's what we should use

**Comment by Princesseuh** [comment]
See my previous comment on this issue and also on the issue you linked for info about that

Using `astro/env` will make the Astro global available in all files even though it doesn't work, which can lead to confusion

**Comment by JuanM04** [comment]
> Using astro/env will make the Astro global available in all files even though it doesn't work, which can lead to confusion. Additionally, the Fragment definition can collide with others. This definition file was exclusively meant for usage in astro files

Ohhhh, I see the problem. We still need to have some sort of `astro/env` (see #3191), maybe could be split up

**Pull request closed without merging**